packages/http/src/util/runtimeExpression.ts

```diff
@@ -0,0 +1,58 @@
+import { IHttpRequest, IHttpResponse } from '../types';
+import * as Option from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { get as _get } from 'lodash';
+import { pointerToPath } from '@stoplight/json';
+
+export function resolveRuntimeExpression(expr: string, request: IHttpRequest, response: IHttpResponse): Option.Option<unknown> {
+  const parts = expr.split(/[.#]/);
+  return pipe(
+    pipe(
+      parts[0],
+      Option.fromPredicate(part => part === '$method'),
+      Option.map(() => String(request.method)),
+    ),
+    Option.alt(() => pipe(
+      parts[0],
+      Option.fromPredicate(part => part === '$statusCode'),
+      Option.map(() => String(response.statusCode)),
+    )),
+    Option.alt(() => pipe(
+      parts[0],
+      Option.fromPredicate(part => part === '$request'),
+      Option.chain(() => pipe(
+        pipe(
+          parts[1],
+          Option.fromPredicate(part => part === 'header'),
+          Option.chain(() => Option.fromNullable(request.headers && request.headers[parts[2]])),
+        ),
+        Option.alt(() => pipe(
+          parts[1],
+          Option.fromPredicate(part => part === 'query'),
+          Option.chain(() => Option.fromNullable(request.url.query && request.url.query[parts[2]])),
```

Comment:
Maybe we could organise this as a series of functions so that the final pipe results a little bit easier to read?

```ts
  return pipe(
    readMethod(),
    Option.alt(() => readStatus()),
    Option.alt(() => readRequest()),
    Option.alt(() => readResponse()),
);
```

and then the single functions will effectively do the stuff.

Reply:
Done 👍